src/main.cc

```diff
@@ -1099,8 +1107,9 @@ ParseFloatOption(const std::string arg)
 }
 #endif
 
+template <>
 double
-ParseDoubleOption(const std::string arg)
+ParseOption(const std::string& arg)
```

Comment:
Won't `ParseOption<int>(optarg)` still return a double?

Reply:
There is already a template specialization for `int` above, so it should get resolved to that implementation.